syncing/src/test/kotlin/maru/syncing/beaconchain/CLSyncServiceImplTest.kt

```diff
@@ -0,0 +1,493 @@
+import maru.crypto.Hashing
+import maru.database.BeaconChain
+import maru.database.InMemoryBeaconChain
+import maru.extensions.fromHexToByteArray
+import maru.p2p.P2PNetworkImpl
+import maru.p2p.PeerLookup
+import maru.p2p.messages.StatusMessageFactory
+import maru.serialization.ForkIdSerializers
+import maru.serialization.rlp.RLPSerializers
+import maru.syncing.beaconchain.pipeline.BeaconChainDownloadPipelineFactory.Config
+import net.consensys.linea.metrics.Counter
+import net.consensys.linea.metrics.MetricsFacade
+import org.apache.tuweni.bytes.Bytes32
+import org.assertj.core.api.Assertions.assertThat
+import org.awaitility.Awaitility.await
+import org.hyperledger.besu.crypto.KeyPair
+import org.hyperledger.besu.crypto.SignatureAlgorithm
+import org.hyperledger.besu.crypto.SignatureAlgorithmFactory
+import org.hyperledger.besu.ethereum.core.Util
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.times
+import org.mockito.kotlin.any
+import org.mockito.kotlin.doAnswer
+import org.mockito.kotlin.never
+import org.mockito.kotlin.reset
+import org.mockito.kotlin.spy
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import tech.pegasys.teku.networking.p2p.libp2p.MultiaddrPeerAddress
+
+class CLSyncServiceImplTest {
+  companion object {
+    private const val CHAIN_ID = 1337u
+    private const val IPV4 = "127.0.0.1"
+    private const val PEER_ID_NODE_2: String = "16Uiu2HAmVXtqhevTAJqZucPbR2W4nCMpetrQASgjZpcxDEDaUPPt"
+
+    private val key1 = "0x0802122012c0b113e2b0c37388e2b484112e13f05c92c4471e3ee1dfaa368fa5045325b2".fromHexToByteArray()
+    private val key2 =
+      "0x0802122100f3d2fffa99dc8906823866d96316492ebf7a8478713a89a58b7385af85b088a1"
+        .fromHexToByteArray()
+  }
+
+  private var port1: UInt = 0u
+  private var port2: UInt = 0u
+  private lateinit var signatureAlgorithm: SignatureAlgorithm
+  private lateinit var keypair: KeyPair
+  private lateinit var beaconChain1: BeaconChain
+  private lateinit var beaconChain2: BeaconChain
+  private lateinit var validators: Set<Validator>
+  private lateinit var p2pNetwork1: P2PNetworkImpl
+  private lateinit var p2pNetwork2: P2PNetworkImpl
+  private lateinit var clSyncService: CLSyncServiceImpl
+  private lateinit var peerLookup: PeerLookup
+
+  @BeforeEach
+  fun setUp() {
+    signatureAlgorithm = SignatureAlgorithmFactory.getInstance()
+    keypair = signatureAlgorithm.generateKeyPair()
+    validators = setOf(Validator(Util.publicKeyToAddress(keypair.publicKey).toArray()))
+
+    val genesisTimestamp = DataGenerators.randomTimestamp()
+    val (genesisBeaconState, genesisBeaconBlock) = genesisState(genesisTimestamp, validators)
+    beaconChain1 = spy(InMemoryBeaconChain(genesisBeaconState, genesisBeaconBlock))
+    beaconChain2 = spy(InMemoryBeaconChain(genesisBeaconState, genesisBeaconBlock))
```

Comment:
I would call it the `sourceBeaconChain`. The `beaconChain1` is then the the target